checker/src/main/java/org/checkerframework/checker/regex/RegexAnnotatedTypeFactory.java

```diff
@@ -487,17 +487,20 @@ public static int getGroupCount(@Regex String regexp) {
         // TODO check and ask for missing cases or alternate ways.
         List<Integer> nonNullGroups = new ArrayList<>();
         int n = getGroupCount(regexp);
-        for (int i = 0; i <= n; i++) {
+        for (int i = 1; i <= n; i++) {
             nonNullGroups.add(i);
         }
         ArrayDeque<Integer> openingIndices = new ArrayDeque<>();
         int group = 0;
         boolean squareBracketOpen = false;
         boolean escaped = false;
```

Comment:
This means that the previous character was `\`.  Writing that comment would be useful.